src/pages/signup-page/SignUpPage.js

```diff
@@ -0,0 +1,201 @@
+import { React, useState } from "react";
+import "./signuppage.css";
+import "../login-page/loginpage.css";
+import { Link, useNavigate } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import { signup } from "../../features/auth/authSlice";
+
+export function SignUpPage() {
+  const [userData, setUserData] = useState({
+    email: "",
+    password: "",
+    confirmPassword: "",
+    firstName: "",
+    lastName: "",
+  });
+  const [error, setError] = useState("");
+  let navigate = useNavigate();
+  const dispatch = useDispatch();
+  const [type, setType] = useState("password");
+  const [confirmPasswordType, setConfirmPasswordType] = useState("password");
+  const [passwordError, setPasswordError] = useState("");
+  const doValidate = () => {
+    if (
+      !/^([A-Za-z0-9_\-\.])+\@([A-Za-z0-9_\-\.])+\.([A-Za-z]{2,4})$/.test(
+        userData.email
+      )
+    ) {
+      setError("Please enter a valid email address");
+      return false;
+    } else {
+      setError("");
+    }
+    if (userData.password !== userData.confirmPassword) {
+      setPasswordError("Password does not match!");
+      return false;
+    } else {
+      setPasswordError("");
+    }
+    return true;
+  };
```

Comment:
you can put this function in utils or helper

Reply:
done